### PR TITLE
GEP 10 - ContainerD Enablement

### PR DESCRIPTION
**Pull request by nimrodoron**
Enablement of ContainerD according to GEP-10

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:
NONE

Format of block header: <category> <target_group>
Possible values:
- category:       improvement|noteworthy|action
- target_group:   user|operator|developer
-->
```improvement operator

```

### CODE REVIEW

File: pkg/generator/templates/cloud-init/suse-jeos.template [hunk 1]
```diff
@@ -27,12 +38,21 @@ write_files:
 {{- end }}
 runcmd:
 - "until zypper -q install -y docker wget socat jq nfs-client; [ $? -ne 7 ]; do sleep 1; done"
+{{ if .IsContainerDEnabled -}}
+- zypper addrepo -G http://download.opensuse.org/update/openSUSE-stable upstream
```
Comment (vpnachev) [inline pkg/generator/templates/cloud-init/suse-jeos.template]: is this avoidable ?
In the GEP we were discussing a docker image to be the source of additional packages, right?

Comment (nimrodoron) [reply pkg/generator/templates/cloud-init/suse-jeos.template]: Currently no.
Only in SuSe the OS image which is currently being used doesn't contain the ctr cli.
I'm using this cli to monitor containerd.
Few options avaliable here:
1. Download and install the ctr-cli.
2. Don't support the ContainerD in SuSe.
3. Don't activate the ContainerD monitor health in SuSe.
4. Replace the SuSe image to a one contains the ctr-cli together with Docker (Exist in SuSe official repository)

Comment (rfranzke) [reply pkg/generator/templates/cloud-init/suse-jeos.template]: @vpnachev Does it make sense to get in touch with the SUSE colleagues and ask them to add this binary?

Comment (vpnachev) [reply pkg/generator/templates/cloud-init/suse-jeos.template]: Yes, I will do it.

**Pull request closed without merging**